hack/update-deps.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -e
+
+glide update
+
+
+#  all the ugorji's need to agree or we end up failing.  Yours, mine, and ours fail!
+rm -rf ./vendor/k8s.io/api/vendor/github.com/ugorji
+rm -rf ./vendor/k8s.io/client-go/vendor/github.com/ugorji
+rm -rf ./vendor/k8s.io/apiserver/vendor/github.com/ugorji
+rm -rf ./vendor/k8s.io/apimachinery/vendor/github.com/ugorji
+
+#  kube-openapi and apiserver need to agree on github.com/emicklei/go-restful or we fail to build.  Yours, mine, and ours fail!
+rm -rf ./vendor/k8s.io/apiserver/vendor/github.com/emicklei
+
+#  our package and apiserver have to agree on github.com/spf13/pflag or we fail to build.  Yours, mine, and ours fail!
+rm -rf ./vendor/k8s.io/apiserver/vendor/github.com/spf13/pflag
```

Comment:
This is interesting. For comparison here's the script we're using to vendor for all our repos via Dan Mace: https://github.com/openshift/online-archivist/blob/master/hack/update-vendor.sh

Reply:
> This is interesting. For comparison here's the script we're using to vendor for all our repos via Dan Mace: https://github.com/openshift/online-archivist/blob/master/hack/update-vendor.sh

Vendoring origin is a difficult beast.  This *should* be easier.  I'd recommend trying to play nice with the tools if possible.  Allowing subvendors in particular can make your dependency graph easier to traverse.